pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -541,12 +541,21 @@ func (n *NetworkPolicyController) processNetworkPolicy(np *networkingv1.NetworkP
 			Services:  toAntreaServices(egressRule.Ports),
 		})
 	}
+	directions := make([]networkpolicy.Direction, 0, len(np.Spec.PolicyTypes))
```

Comment:
do we know for sure that np.Spec.PolicyTypes will be always populated? or we also have to infer from the presence of the ingress or egress rules and set the direction accordingly?

Reply:
The PolicyTypes is only used to decide whether to isolate traffic (add default-deny rule) when the rules in a direction is not present. If ingress rule exists, ingress must be isolated, if egress exists, egress must be isolated.